spec/models/lettings_log_spec.rb

```diff
@@ -250,1841 +251,236 @@
-            end
-          end
-
-          context "when rent_type is Rent to Buy" do
-            let(:rent_type) { 3 }
-
-            it "derives the most recent let type as Intermediate Rent basis if it is a renewal and clears it if it is not" do
-              expect { lettings_log.update!(renewal: 1) }.to change(lettings_log, :unitletas).to 4
-              expect { lettings_log.update!(renewal: 0) }.to change(lettings_log, :unitletas).from(4).to nil
-            end
-          end
-
-          context "when rent_type is London Living Rent" do
-            let(:rent_type) { 4 }
-
-            it "derives the most recent let type as Intermediate Rent basis if it is a renewal and clears it if it is not" do
-              expect { lettings_log.update!(renewal: 1) }.to change(lettings_log, :unitletas).to 4
-              expect { lettings_log.update!(renewal: 0) }.to change(lettings_log, :unitletas).from(4).to nil
-            end
-          end
-
-          context "when rent_type is Other intermediate rent product" do
-            let(:rent_type) { 5 }
-            let(:irproduct_other) { "Rent first" }
-
-            it "derives the most recent let type as Intermediate Rent basis if it is a renewal and clears it if it is not" do
-              expect { lettings_log.update!(renewal: 1) }.to change(lettings_log, :unitletas).to 4
-              expect { lettings_log.update!(renewal: 0) }.to change(lettings_log, :unitletas).from(4).to nil
-            end
-          end
-        end
-
-        context "when collection year is 2023/24 or later" do
-          let(:now) { Time.zone.local(2024, 1, 1) }
-
-          context "when rent_type is Social Rent" do
-            let(:rent_type) { 0 }
+      )
+    end
 
-            it "derives the most recent let type as Social Rent basis if it is a renewal and clears it if it is not" do
-              expect { lettings_log.update!(renewal: 1) }.to change(lettings_log, :unitletas).to 1
-              expect { lettings_log.update!(renewal: 0) }.to change(lettings_log, :unitletas).from(1).to nil
-            end
-          end
+    it "correctly derives and saves partial and full major repairs date" do # weird test, am I missing something?
+      record_from_db = described_class.find(lettings_log.id)
+      expect(record_from_db["mrcdate"].day).to eq(4)
+      expect(record_from_db["mrcdate"].month).to eq(5)
+      expect(record_from_db["mrcdate"].year).to eq(2021)
+    end
 
-          context "when rent_type is Affordable Rent" do
-            let(:rent_type) { 1 }
+    it "correctly derives and saves day, month, year from start date" do # weird test, am I missing something?
+      record_from_db = described_class.find(lettings_log.id)
+      expect(record_from_db["startdate"].day).to eq(10)
+      expect(record_from_db["startdate"].month).to eq(10)
+      expect(record_from_db["startdate"].year).to eq(2021)
+    end
```

Comment:
sanity check please. These tests look silly to me, we save a date in the db, we call it back out, we can do standard ruby date things to it.
Can I delete?